### PR TITLE
Security group updates

### PR DESCRIPTION
**Pull request by danjbh**
*NOTE: The work in this PR mirrors the security group updates performed in cloudposse/terraform-aws-elasticache-redis#133, since these modules are very similar.*

## what
- Upgrade to new Cloud Posse security group standards
- Enable `create_before_destroy` on security groups by default
- Update AWS provider version constraint to properly require a version that has all the features the module uses
- Add security group ARN and name to `outputs.tf`

## why
- Further standardize all Cloud Posse modules
- Supply bug fixes and features requested via "issues".

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -22,45 +22,56 @@ resource "null_resource" "cluster_urls" {
...
-  cidr_blocks       = ["0.0.0.0/0"]
-  security_group_id = join("", aws_security_group.default.*.id)
-  type              = "egress"
-}
+module "aws_security_group" {
+  source  = "cloudposse/security-group/aws"
+  version = "0.4.3"
+
+  enabled = local.create_security_group
+
```
Comment (nitrocode) [inline main.tf:47]: no description?

File: security-group-inputs.tf [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  description = <<-EOT
+    A list of IDs of Security Groups to allow access to the security group created by this module.
+  EOT
+}
+
+locals {
+  allowed_security_group_ids = concat(var.allowed_security_groups, var.allowed_security_group_ids)
+}
+
+
```
Comment (nitrocode) [inline security-group-inputs.tf:40]:
```suggestion
```

**Comment by nitrocode** [comment]
Is the security group inputs the latest export from the sg module?

Were you able to upgrade an existing customer to this prs module?

Could you include a migration guide like we did in previous modules that updated to sh 4.x?

**Review by Nuru — APPROVED** [review]

**Pull request merged**